packages/blockchain-wallet-v4-frontend/src/modals/BuySell/UpdateSecurityCode/UpdateSecurityCode.tsx

```diff
@@ -0,0 +1,171 @@
+    }
+  }
+
+  useEffect(() => {
+    if (cvvHasData) {
+      setTimeout(() => {
+        dispatch(actions.components.buySell.setStep({ step: 'CHECKOUT_CONFIRM' }))
+      }, 500)
+    }
+  }, [cvvHasData, dispatch])
+
+  if (!method) {
+    backToEnterAmount()
+    return null
+  }
+
+  return (
+    <FlyoutContainer>
+      <FlyoutHeader data-e2e='update-security-code-flyout' mode='back' onClick={backToEnterAmount}>
+        <FormattedMessage id='buttons.security' defaultMessage='Security' />
+      </FlyoutHeader>
+      <FlyoutContent mode='middle'>
+        <Flex flexDirection='column'>
+          <Padding horizontal={2.5}>
+            <Flex justifyContent='center'>
+              <div style={{ transform: 'scale(2.5) translateY(-1rem)' }}>
+                <Logo
+                  doubleVariant='badge'
+                  primaryContent={{
+                    altText: 'Credit Card Icon',
+                    backgroundColor: PaletteColors['grey-100'],
+                    icon: <IconCreditCard color={PaletteColors['grey-900']} size='medium' />
+                  }}
+                  secondaryContent={{
+                    altText: 'Shield Icon',
+                    backgroundColor: PaletteColors['white-000'],
+                    icon: <IconShield color={SemanticColors.body} size='small' />
+                  }}
+                  size='large'
+                />
+              </div>
+            </Flex>
+            <Text as='p' variant='title2' textAlign='center'>
+              <FormattedMessage id='copy.security_code' defaultMessage='Security Code' />
+            </Text>
+            <Padding horizontal={1.5} bottom={2.5}>
+              <Text as='p' variant='body1' textAlign='center' color={SemanticColors.overlay}>
+                <FormattedMessage
+                  id='copy.re_enter_security_code'
+                  defaultMessage='Please re-enter the 3 digit CVV code associated with the card below'
+                />
+              </Text>
+            </Padding>
+            <Padding bottom={1}>
+              <div style={{ height: '97.5px' }}>
+                <Input
+                  id='primary'
+                  label='CVV Code'
+                  placeholder='000'
+                  state={cvvHasError ? 'error' : 'default'}
+                  type='number'
+                  // @ts-ignore
+                  autoComplete='cc-csc'
+                  aria-label='Card Security Code'
+                  aria-placeholder='CVC'
+                  postfix={
+                    cvvHasData ? (
+                      <IconCheckCircle color={SemanticColors.success} size='medium' />
+                    ) : (
+                      <IconLockClosed color={SemanticColors.muted} size='medium' />
+                    )
+                  }
+                  onChange={(e) => setCvv(e.currentTarget.value)}
+                  helperText={
+                    cvvHasError && (
```

Comment:
Question here - if `cvvHasError` is `null`/`undefined`/`false` it will return that value; can `helperText` parse those value correctly in the error cases?

Reply:
Fair point. It takes a react node so would be fine in case of null but perhaps not false. I'll be more explicit and pass `<></>`